rust/rope/src/tree.rs

```diff
@@ -763,25 +713,35 @@ impl<'a, N: NodeInfo> Cursor<'a, N> {
         CursorIter { cursor: self, _metric: PhantomData }
     }
 
+    /// Tries to find the last boundary in the leaf the cursor is currently in.
+    ///
+    /// If the last boundary is at the end of the leaf, it is only counted if
+    /// it is less than `orig_pos`.
```

Comment:
to clarify: If we're at the 0 offset in leaf2, and we call this function, we want to avoid returning `leaf1.len()`, which would be the same as our starting offset?

Is that the only case that we're dealing with, here?

Reply:
Yes, that's the motivation. The `prev` method always yields a position strictly less than the starting position.